index.js

```diff
@@ -21,8 +21,8 @@ module.exports = {
   ],
   extends: [
     "eslint:recommended",
-    "plugin:@typescript-eslint/recommended",
-    "plugin:@typescript-eslint/recommended-requiring-type-checking",
+    "plugin:@typescript-eslint/recommended-type-checked",
```

Comment:
Renamed to match the `eslint` docs, `requiring-type-checking` appears to be old syntax. Since we serialize the config in our tests, we're confident there's no regression.